### ISSUE TITLE
Onslaught 2 - regression

### ISSUE THREAD

**Issue by eiglow** [lightspark/lightspark#498]
swf: [Onslaught2.zip](https://github.com/lightspark/lightspark/files/4566984/Onslaught2.zip)

Unfortunately, my testing shows that commit d149e9d2a13beaa4390b94bba46de0fb832d7480 seems to have regressed playback of this swf. Before, it would show an Unsupported error. After, Lightspark segfaults.

Here is a full backtrace: [bt.log](https://github.com/lightspark/lightspark/files/4566991/bt.log)

**Comment by eiglow** [comment]
The same commit also causes a segfault with [srpgcompletexgen.zip](https://github.com/lightspark/lightspark/files/4566999/srpgcompletexgen.zip). Here's the backtrace from that game: [bt.log](https://github.com/lightspark/lightspark/files/4567004/bt.log)

**Comment by masatake** [comment]
> Unfortunately, my testing shows that commit d149e9d seems to have regressed playback of this swf. Before, it would show an Unsupported error. After, Lightspark segfaults.

Isn't is possible to detect the regression by utilizing travis-ci or something?

**Comment by eiglow** [comment]
Perhaps, maybe we could have a script run a few known-working SWFs and make sure they don't crash Lightspark. It'd help even more if maybe when Lightspark is run with a certain parameter, Unsupported errors could return a certain exit code, which would make it possible to write tests for them. I've never written tests or anything before though, so I could be wrong about all this.

**Comment by dbluelle** [comment]
I see some problems with running "real" swf files for automated testing:
- While it's currently possible to run a swf file that will exit on error (parameter --exit-on-error, we already use that for the tamarin test suite), how do we determine if the swf is ok?
I think we have to set some time limit for every swf after which we kill the process and assume the swf is ok.
- What about swfs that crash only after some user interaction (press a play button etc.) ?
I don't know how to automatically execute some specific mouse clicks at a specific time...
Maybe by creating some kind of script language, but that would be a big task...

**Comment by masatake** [comment]
Just for detecting a crash, I guess we don't need a perfect one.

It seems that Gnash carried out such a "big task".

http://git.savannah.gnu.org/cgit/gnash.git/tree/testsuite?id=583ccbc1275c7701dc4843ec12142ff86bb305b4

**Comment by dbluelle** [comment]
Well, if anyone creates a similar tool for lightspark, I will be happy to add it to this repository.

**Comment by masatake** [comment]
We have to inspect the result of Gnash carefully because there is a "license gap" between Gnash and lightspark.

**Comment by eiglow** [comment]
Looks like commit cf9b6fd6f913882ea26c5ff75f989fabc952503f fixed the segfault. We're back to an unsupported error on this swf; I will open a separate issue for that.
